crates/autopilot/src/run_loop.rs

```diff
@@ -310,7 +309,14 @@ impl RunLoop {
 
     /// Runs the solver competition, making all configured drivers participate.
     async fn competition(&self, id: AuctionId, auction: &Auction) -> Vec<Participant<'_>> {
-        let fee_policies = fee::Policies::new(auction, self.fee_policy.clone());
+        let fee_policies = self
+            .policy_factory
+            .build(auction)
+            .await
+            .unwrap_or_else(|err| {
```

Comment:
I think this error handling is a bit too strict. While I agree we should investigate whenever we cannot attach a fee to an order as planned, we don't need to nullify all fees for all other orders that worked out fine.

Reply:
Fixed in the policy factory.